### PR TITLE
#1322 redesigned project details section and created teams dropdown

### LINKED ISSUE #1322: Project Form Redesign - Redesign Details Section
### Description

Redesign the Details Section so that it looks in accordance with the new design. 

### Acceptance Criteria

Looks like Mockup

### Proposed Solution

will definitely remove the pageblock. would suggest using a grid layout for it so its easy to manage smaller screens. Id use project form labels for the headers. Also team is acutally Teams and should be an array of teams. 

Work in the file Project Edit Details, but rename the component to be ProjectFormDetails and rename the file to be the same. 

Work off of the feature/Project-Form-Redesign Feature Branch

### Mocks

Old Version
Screen Shot 2023-09-05 at 6 14 48 PM

New Version
Screen Shot 2023-09-05 at 6 14 32 PM

### PR DESCRIPTION
**Pull request by Aaryan1203**
Reformatted project edit details section
Created a teams dropdown form

Change Request dropdown no longer works

## Screenshots
image

Smaller Screen:
image

## Checklist

It can be helpful to check the `Checks` and `Files changed` tabs.
Please review the [contributor guide](https://nerdocs.atlassian.net/wiki/spaces/NER/pages/8060929/Software+Contributor+Guide) and reach out to your Tech Lead if anything is unclear.
Please request reviewers and ping on slack only after you've gone through this whole checklist.

- [x] All commits are tagged with the ticket number
- [x] No linting errors / newline at end of file warnings
- [x] All code follows repository-configured prettier formatting
- [x] No merge conflicts
- [x] All checks passing
- [x] Screenshots of UI changes (see Screenshots section)
- [x] Remove any non-applicable sections of this template
- [x] Assign the PR to yourself
- [x] No `yarn.lock` changes (unless dependencies have changed)
- [ ] Request reviewers & ping on Slack
- [ ] PR is linked to the ticket (fill in the closes line below)

Closes #1322

### CODE REVIEW

**Review by ninest — APPROVED** [review]
All changes have been copied from the other PR because it was on the wrong branch

File: src/frontend/src/pages/ProjectDetailPage/ProjectEdit/ProjectEditDetails.tsx [hunk 1]
```diff
@@ -33,9 +34,12 @@ const ProjectEditDetails: React.FC<ProjectEditDetailsProps> = ({
...
   return (
-    <PageBlock title="Project Details">
-      <Grid container spacing={2}>
-        <Grid item xs={8}>
+    <Box>
+      <Typography variant="h5" sx={{ marginBottom: '10px' }}>
+        Project Details
+      </Typography>
+      <Grid container spacing={3}>
+        <Grid item lg={2.4} xs={12}>
```
Comment (RChandler234) [inline src/frontend/src/pages/ProjectDetailPage/ProjectEdit/ProjectEditDetails.tsx:42]: you should use an md between lg and xs, it switches to the stacked view too quickly rn

File: src/frontend/src/pages/ProjectDetailPage/ProjectEdit/ProjectEditDetails.tsx [hunk 2]
```diff
@@ -70,7 +110,7 @@ const ProjectEditDetails: React.FC<ProjectEditDetailsProps> = ({
             value={userToAutocompleteOption(users.find((user) => user.userId.toString() === projectLead))}
           />
         </Grid>
-        <Grid item xs={12} md={6} sx={{ mt: 1 }}>
+        <Grid item xs={12} md={3} sx={{ mt: 1 }}>
           <FormLabel>Project Manager</FormLabel>
           <NERAutocomplete
```
Comment (RChandler234) [inline src/frontend/src/pages/ProjectDetailPage/ProjectEdit/ProjectEditDetails.tsx:115]: you should change the NERAutocomplete to look like it does in this PR: https://github.com/Northeastern-Electric-Racing/FinishLine/pull/1682 (ignore all the files except for NERAutocomplete)

**Review by RChandler234 — CHANGES_REQUESTED** [review]

**Review by RChandler234 — APPROVED** [review]
LGTM 🔥

**Pull request merged**